cmd/util/output/output.go

```diff
@@ -0,0 +1,122 @@
+package output
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/samber/lo"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v2"
+)
+
+type OutputFormat string
+
+const (
+	TableFormat OutputFormat = "table"
+	CSVFormat   OutputFormat = "csv"
+	JSONFormat  OutputFormat = "json"
+	YAMLFormat  OutputFormat = "yaml"
+)
+
+var AllFormats = []OutputFormat{TableFormat, CSVFormat, JSONFormat, YAMLFormat}
+
+var noStyle = table.Style{
+	Name:   "StyleDefault",
+	Box:    table.StyleBoxDefault,
+	Color:  table.ColorOptionsDefault,
+	Format: table.FormatOptionsDefault,
+	HTML:   table.DefaultHTMLOptions,
+	Options: table.Options{
+		DrawBorder:      false,
+		SeparateColumns: false,
+		SeparateFooter:  false,
+		SeparateHeader:  false,
+		SeparateRows:    false,
+	},
+	Title: table.TitleOptionsDefault,
+}
+
+type OutputOptions struct {
+	Format     OutputFormat // The output format for the list of jobs
+	HideHeader bool         // Hide the column headers
+	NoStyle    bool         // Remove all styling from table output.
+	Wide       bool         // Print full values in the table results
+}
+
+type TableColumn[T any] struct {
+	table.ColumnConfig
+	Value func(T) string
+}
+
+func Output[T any](cmd *cobra.Command, columns []TableColumn[T], options OutputOptions, items []T) error {
+	switch options.Format {
+	case JSONFormat:
+		return json.NewEncoder(cmd.OutOrStdout()).Encode(items)
+	case YAMLFormat:
+		return yaml.NewEncoder(cmd.OutOrStdout()).Encode(items)
+	case CSVFormat:
+		fallthrough
+	case TableFormat:
+		outputTable[T](cmd, columns, options, items)
+		return nil
```

Comment:
nit: I find `case CSVFormat, TableFormat:` clearer as a reader and typically use `fallthrough` when taking some action before falling through to the next case. Not a big deal though.

Reply:
Ah, I didn't know you could include multiple values like this. Thanks!